### DISCUSSION TITLE
[Feature] <Migration>

### DISCUSSION THREAD

**Discussion by grossmanns** [immich-app/immich#7468] [Q&A]
### The feature

Impossible to migrate from old to new server. What I did :

- I created a new docker stack, like written in the doc. Did not go to the ip:2283
- I copied all files from old server to the new one
- on the old server : docker exec -t immich_postgres pg_dumpall -c -U postgres | gzip > "/media/backup-database/dump.sql.gz"
- copy of that dump to the new server
- docker compose down -v
- docker compose pull
- docker compose create
- docker start immich_postgres
- gunzip < "/media/database/dump.sql.gz" | docker exec -i immich_postgres psql -U postgres -d immich

And I got a lot of errors : 
`ERROR:  function "f_concat_ws" already exists with same argument types
ALTER FUNCTION
SET
SET
ERROR:  relation "activity" already exists
ALTER TABLE
ERROR:  relation "albums" already exists
ALTER TABLE
COMMENT
ERROR:  relation "albums_assets_assets" already exists
ALTER TABLE
ERROR:  relation "albums_shared_users_users" already exists
ALTER TABLE
ERROR:  relation "api_keys" already exists
ALTER TABLE
ERROR:  type "public.vector" does not exist`

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by schuhbacca** [comment]
When you created the new stack, did you start it once before restoring the DB to it?

**Reply by grossmanns** [reply]
no

**Comment by alextran1502** [comment]
What was the version of Immich you had on the old instance?

**Reply by grossmanns** [reply]
it is the [v1.92.1]

**Comment by grossmanns** [comment]
Just recreatet the whole container (proxmox) and here is the beginning of the log : 

`root@immich:/media# gunzip < "/media/backup-database/dump.sql.gz" | docker exec -i immich_postgres psql -U postgres -d immich
SET
SET
SET
ERROR:  cannot drop the currently open database
ERROR:  current user cannot be dropped
ERROR:  role "postgres" already exists
ALTER ROLE
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
UPDATE 1
DROP DATABASE
CREATE DATABASE
ALTER DATABASE
You are now connected to database "template1" as user "postgres".
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
COMMENT
ALTER DATABASE
You are now connected to database "template1" as user "postgres".
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
REVOKE
GRANT
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
ERROR:  database "immich" already exists
ALTER DATABASE
You are now connected to database "immich" as user "postgres".
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
CREATE EXTENSION
COMMENT
CREATE EXTENSION
COMMENT
CREATE EXTENSION
COMMENT
ERROR:  extension "vectors" must be installed in schema "vectors"
ERROR:  extension "vectors" does not exist
ERROR:  function "f_concat_ws" already exists with same argument types
ALTER FUNCTION
SET
SET
ERROR:  relation "activity" already exists
ALTER TABLE
ERROR:  relation "albums" already exists
ALTER TABLE
COMMENT
ERROR:  relation "albums_assets_assets" already exists
ALTER TABLE
ERROR:  relation "albums_shared_users_users" already exists
ALTER TABLE
ERROR:  relation "api_keys" already exists
ALTER TABLE
ERROR:  type "public.vector" does not exist
LINE 4:     embedding public.vector(512) NOT NULL,
                      ^
ERROR:  relation "public.asset_faces" does not exist
ERROR:  relation "asset_job_status" already exists
ALTER TABLE
ERROR:  relation "assets" already exists
ALTER TABLE`

**Comment by nickaldred** [comment]
Did you manage to fix this, im getting the same issue?

**Reply by grossmanns** [reply]
No. Had to reinstall everything. Lost an afternoon...
